src/renderers/shared/fiber/ReactFiberStack.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberStack
+ * @flow
+ */
+
+'use strict';
+
+import type { Fiber } from 'ReactFiber';
+
+export type StackCursor = {|
+  current: any
+|};
+
+const emptyObject = require('emptyObject');
+const warning = require('warning');
+
+const valueStack : Array<Object> = [];
+
+if (__DEV__) {
+  var fiberStack : Array<Fiber | null> = [];
+}
+
+let index = -1;
+
+exports.createCursor = function() : StackCursor {
```

Comment:
Passing a default value feels a bit awkward since (at least for now) none of the cursors have meaningful defaults. But the generics benefit is nice! So sure! 😄